sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/OrderbyDocumentQueryTest.java

```diff
@@ -230,6 +241,49 @@ public void queryOrderByString() throws Exception {
         validateQuerySuccess(queryObservable.byPage(pageSize), validator);
     }
 
+    @Test(groups = { "simple" }, timeOut = TIMEOUT, dataProvider = "sortOrder")
+    public void queryOrderByMixedTypes(String sortOrder) throws Exception {
+        List<PartitionKeyRange> partitionKeyRanges = getPartitionKeyRanges(createdCollection.getId(),
+                                                                           BridgeInternal.getContextClient(this.client));
+        // Ensure its a cross partition query
+        assertThat(partitionKeyRanges.size()).isGreaterThan(1);
+        // We are inserting documents with int, float, string, array, object and missing propMixed.
+        String query = String.format("SELECT r.id, r.propMixed FROM r ORDER BY r.propMixed ", sortOrder);
+        CosmosQueryRequestOptions options = new CosmosQueryRequestOptions();
+        List<String> orderedIds = createdDocuments.stream()
+                                      .sorted((o1, o2) -> ItemComparator.getInstance().compare(o1.get("propMixed"),
```

Comment:
your test validation relies on `ItemComparator` which essentially means the validation uses the existing implementation comparator to validate the expected order.

This is not ideal as if I go and change ItemComparator in the implementation and introduce a bug or change the comparer and its ordering, still this test will pass.

For the validation you shouldn't have dependency on the implementation.

Please try to find the expected order differently without relying on the implementation.

Reply:
Yes this is something I am aware of. In the end I am not using the ordering at all for validation and just checking to make sure all the documents are present.

Reply:
Removed ItemComparator and also improved the test to check for proper grouping of results and sorting inside each group